### ISSUE TITLE
Add new CSS class for available_option

### ISSUE THREAD

**Issue by jaimeiniesta** [maxmarcon/live_select#12]
I'm trying to add a different CSS style for the selected options, so in the dropdown the cursor is changed to "none", but I can't see how to do this. This is my current setup:

```elixir
<%= live_select f,
                  :tags,
                  mode: :tags,
                  options: @live_select_options,
                  update_min_len: 1,
                  placeholder: "Add tag",
                  style: :none,
                  container_class: "tag_editor_container",
                  text_input_class: "tag_editor_text_input",
                  dropdown_class: "tag_editor_dropdown",
                  option_class: "tag_editor_option",
                  active_option_class: "tag_editor_active_option",
                  selected_option_class: "tag_editor_selected_option" %>
```

```css
@layer components {
  .tag_editor_container {
    @apply border my-4 p-4 rounded-md bg-gray-100;
  }

  .tag_editor_text_input {
    @apply  bg-gray-100 text-gray-900;
  }

  .tag_editor_text_input:focus {
    @apply outline-none;
  }

  .tag_editor_dropdown {
    @apply pt-4 pb-2 space-y-2;
  }

  .tag_editor_option {
    @apply cursor-pointer;
  }

  .tag_editor_option:hover {
    @apply font-bold text-blue-800;
  }

  .tag_editor_active_option {
    @apply font-bold text-blue-800;
  }

  .tag_editor_selected_option {
    @apply font-normal text-gray-500 cursor-default;
  }
}
```

I want to make it clearer that the options that have been already selected are no longer clickable, so that's why I'd like to avoid a pointer cursor in them.

Maybe we could have an "available_option" class, as the opposite and complementary class to "selected_option"? This way we could style the options depending on that condition, available or already selected.

**Comment by maxmarcon** [comment]
Oh you're styling everything yourself from scratch? You're brave! 👍 Would be interest in seeing the end result.

So you want to have `cursor-pointer` on the available options and `cursor-default` on the already selected ones?

Try doing this: 

```elixir
live_select(my_form, city_search,
   dropdown_class: "cursor-pointer",
   mode: :tags,
   placeholder: "Search for a city",
   selected_option_class: "cursor-default",
   style: :none
)
```

basically, using `cursor-pointer` in the `dropdown_class` and not in the `option_class`.

You can try out the above settings here: https://live-select.fly.dev/?active_option_class=&container_class=&debounce=100&disabled=false&dropdown_class=cursor-pointer&field_name=city_search&form_name=my_form&max_selectable=0&mode=tags&option_class=&placeholder=Search+for+a+city&search_delay=10&selected_option_class=cursor-default&show_styles=true&style=none&style_filter=dropd&tag_class=&tags_container_class=&text_input_class=&text_input_selected_class=&update_min_len=

**Comment by jaimeiniesta** [comment]
Thank you!

Yes I'm using my own classes, as out of the box with Tailwind it didn't look right. Maybe it's because I'm using Tailwind 2, or maybe because it collides with other customizations. Also, I'm not using DaisyUI yet. Anyway I want to give a different look.

The tip about the cursors works fine, thanks! I used `cursor-not-allowed` for the `selected_option_class`.

I still want to make another change; the options have a `hover` style, so they change to blue and bold. I would like the selected items not have this hover effect, so they stay in gray, and normal font, so it's clearer that they're not selectable. How would you approach this?

In this screenshot, I have selected a tag so it appears gray below. The other options have a blue bold hover effect:

<img width="306" alt="Captura de pantalla 2023-01-20 a las 8 40 07" src="https://user-images.githubusercontent.com/2629/213642783-87d83a7f-4e98-4c29-9258-f496b7e1799f.png">

But I would like that when I hover over a selected option, it not only changes the cursor, but also stays in gray, normal font:

<img width="260" alt="Captura de pantalla 2023-01-20 a las 8 40 38" src="https://user-images.githubusercontent.com/2629/213642888-2f3801a2-9adf-44f1-b38a-88d2f049af2b.png">

**Comment by maxmarcon** [comment]
> Yes I'm using my own classes, as out of the box with Tailwind it didn't look right. Maybe it's because I'm using Tailwind 2, or maybe because it collides with other customizations. Also, I'm not using DaisyUI yet. Anyway I want to give a different look.

In case you were not aware, I just wanted to point out that you can also _extend_ the default tailwind/daisyUI styles.
You can add classes to the default styles, and you can also remove classes using the `!class_to_remove` notation. So even if the default styles don't look quite right, you can tweak them. It might be easier than redoing everything from scratch.

> But I would like that when I hover over a selected option, it not only changes the cursor, but also stays in gray, normal font:

Unfortunately this is not possible now. the `option_class` takes precedence over `selected_option_class`, which doesn't seem right. If the option is selected, we should probably apply `selected_option_class`  and ignore `option_class`. That should provide enough flexibility, so you could do: 

```elixir
live_select(my_form, city_search,
   mode: :tags,
   option_class: "hover:font-bold hover:text-blue-800 #{common_option_classes}",
   placeholder: "Search for a city",
   selected_option_class: common_option_classes,
   style: :none
)
```

where `common_option_classes` are classes that apply to both selected and unselected options.

Does it make sense?

**Comment by jaimeiniesta** [comment]
Thanks, 

I saw that the CSS classes can be extended, but I found it easier to build everything from scratch in my case.

I'd see it clearer if we had these classes:

* `option_class` for the common class for each option. For example, a rounder border.
* `available_option_class` for the options that can be picked. For example, blue underline when hovered, pointer cursor.
* `selected_option_class` for the options that can't be picked. For example, grayed out and no hover, not-allowed cursor.

The solution you propose can work too, but it's a bit unclear to me, and then we need to have a `common_option_classes` variable outside.

**Comment by maxmarcon** [comment]
Yes that makes sense. I'm not looking forward to adding a new class option but this seems to be a good reason to do it.

so: 

* `option_class`: applied to all options
* `available_option_class`: applied to options that can be selected
* `selected_option_class`: applied to options that are already selected 
* `active_option_class`: applied to the selectable option that the user currently navigated to with the arrow keys

**Comment by jaimeiniesta** [comment]
Sounds good to me! Having these classes makes it easier to customize.

**Comment by maxmarcon** [comment]
I made the changes. You can try out the main branch and see if you find the styling more convenient now

**Comment by jaimeiniesta** [comment]
Thanks! 

I just tried this and works great! 👍

**Comment by maxmarcon** [comment]
closed with 6fb8fa06b15a8427946e79500eabb029e2bf05ee
